crypto/keys/keybase_keyring.go

```diff
@@ -0,0 +1,511 @@
+		}
+
+		priv, err = cryptoAmino.PrivKeyFromBytes([]byte(linfo.PrivKeyArmor))
+		if err != nil {
+			return nil, err
+		}
+
+	case ledgerInfo, offlineInfo, multiInfo:
+		return nil, errors.New("only works on local private keys")
+	}
+
+	return priv, nil
+}
+
+func (kb keyringKeybase) Export(name string) (armor string, err error) {
+	bz, err := kb.db.Get(string(infoKey(name)))
+	if err != nil {
+		return "", err
+	}
+	if bz.Data == nil {
+		return "", fmt.Errorf("no key to export with name %s", name)
+	}
+
+	return mintkey.ArmorInfoBytes(bz.Data), nil
+}
+
+// ExportPubKey returns public keys in ASCII armored format.
+// Retrieve a Info object by its name and return the public key in
+// a portable format.
+func (kb keyringKeybase) ExportPubKey(name string) (armor string, err error) {
+	bz, err := kb.Get(name)
+	if err != nil {
+		return "", err
+	}
+
+	if bz == nil {
+		return "", fmt.Errorf("no key to export with name %s", name)
+	}
+	return mintkey.ArmorPubKeyBytes(bz.GetPubKey().Bytes()), nil
+}
+
+func (kb keyringKeybase) Import(name string, armor string) (err error) {
+	bz, err := kb.Get(name)
+
+	if err == nil {
+		pubkey := bz.GetPubKey()
+
+		if len(pubkey.Bytes()) > 0 {
+			return errors.New("Cannot overwrite data for name " + name)
+		}
+	}
+
+	infoBytes, err := mintkey.UnarmorInfoBytes(armor)
+	if err != nil {
+		return
+	}
+
+	info, err := readInfo(infoBytes)
+
+	if err != nil {
+		return
+	}
+
+	kb.writeInfo(name, info)
+
+	err = kb.db.Set(keyring.Item{
+		Key:  string(addrKey(info.GetAddress())),
+		Data: infoKey(name),
+	})
+
+	if err != nil {
+		return
+	}
+
+	return nil
+}
+
+// ExportPrivKey returns a private key in ASCII armored format.
+// It returns an error if the key does not exist or a wrong encryption passphrase is supplied.
+func (kb keyringKeybase) ExportPrivKey(name string, decryptPassphrase string,
+	encryptPassphrase string) (armor string, err error) {
+	priv, err := kb.ExportPrivateKeyObject(name, decryptPassphrase)
+	if err != nil {
+		return "", err
+	}
+
+	return mintkey.EncryptArmorPrivKey(priv, encryptPassphrase), nil
+}
+
+// ImportPrivKey imports a private key in ASCII armor format.
+// It returns an error if a key with the same name exists or a wrong encryption passphrase is
+// supplied.
+func (kb keyringKeybase) ImportPrivKey(name string, armor string, passphrase string) error {
+	if _, err := kb.Get(name); err == nil {
+		return errors.New("Cannot overwrite key " + name)
+	}
+
+	privKey, err := mintkey.UnarmorDecryptPrivKey(armor, passphrase)
+	if err != nil {
+		return errors.Wrap(err, "couldn't import private key")
+	}
+
+	kb.writeLocalKey(name, privKey, passphrase)
+	return nil
+}
+
+// ImportPubKey imports ASCII-armored public keys.
+// Store a new Info object holding a public key only, i.e. it will
+// not be possible to sign with it as it lacks the secret key.
+//ImportPubKey for keyring
+func (kb keyringKeybase) ImportPubKey(name string, armor string) (err error) {
+	bz, err := kb.Get(name)
```

Comment:
```suggestion
	bz, _ := kb.Get(name)
```

Reply:
actually, this could be cleaned up to a util function to avoid duplicated code:

```go
if HasKey(kb, name) {
 return errors.New("cannot overwrite data for name " + name)
}
```